storage.js

```diff
@@ -148,23 +138,69 @@ function isCheckbox(el) {
 
 function changePref(event) {
 	var el = event.target;
-	var value = isCheckbox(el) ? el.checked : el.value;
-	localStorage[el.id] = value
-	notifyAllTabs({method: "prefChanged", prefName: el.id, value: value});
+	prefs.setPref(el.id, isCheckbox(el) ? el.checked : el.value);
 }
 
 // Accepts a hash of pref name to default value
 function loadPrefs(prefs) {
 	for (var id in prefs) {
-		var value = typeof localStorage[id] == "undefined" ? prefs[id] : localStorage[id];
+		var value = this.prefs.getPref(id);
 		var el = document.getElementById(id);
 		if (isCheckbox(el)) {
-			if (value == "true") {
-				el.checked = true;
-			}
+			el.checked = value;
 		} else {
 			el.value = value;
 		}
 		el.addEventListener("change", changePref);
 	}
 }
+
+var prefs = {
+// NB: localStorage["not_key"] is undefined, localStorage.getItem("not_key") is null
+
+	// defaults
+	"openEditInWindow": false, // new editor opens in a own browser window
+	"show-badge": true,        // display text on popup menu icon
+	"smart-indent": true,      // CodeMirror smart indent
+
+	"popup.breadcrumbs": true, // display "New style" links as URL breadcrumbs
+	"popup.breadcrumbs.usePath": false, // use URL path for "this URL"
+
+	"popup.enabledFirst": true,  // display enabled styles before disabled styles
+	"manage.enabledFirst": true, // display enabled styles before disabled styles
+
+	"observer.observeFrameContent": false, // [hh] add MutationObserver inside IFRAMEs
+	"observer.observeFrameLoad": false,    // [hh] add onLoad listener to IFRAMEs
+	// https://github.com/JasonBarnabe/stylish-chrome/pull/39#issuecomment-76681235
+
+	NO_DEFAULT_PREFERENCE: "No default preference for '%s'",
+	UNHANDLED_DATA_TYPE: "Default '%s' is of type '%s' - what should be done with it?",
+
+	getPref: function(key, ifUndefined) {
+	// Returns localStorage[key], ifUndefined, this[key], or undefined
+	//   as type of ifUndefined, this[key], or localStorage[key]
+		if (ifUndefined === undefined) ifUndefined = this[key]; // default value
+		var value = localStorage[key];
+		if (undefined === value) { // no user preference
+			if (ifUndefined === undefined) console.error(this.NO_DEFAULT_PREFERENCE, key);
```

Comment:
`console.error()` is not `throw`; execution continues after the call [[1]](https://developer.chrome.com/devtools/docs/console-api#consoleerrorobject-object) [[2]](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw). For example,

````javascript
var i = "one"; console.error("Eek! An error!"); i = "two";
var j = "one"; throw "Eek! Another error!"; j = "two";
// i is "two", j is "one"
````